model/task/task_test.go

```diff
@@ -3669,6 +3788,154 @@ func TestArchive(t *testing.T) {
+	defer func() {
+		assert.NoError(t, db.ClearCollections(Collection, OldCollection, event.LegacyEventLogCollection))
+	}()
+
+	assert.NoError(t, db.ClearCollections(Collection, OldCollection, event.LegacyEventLogCollection))
+	t1 := Task{
+		Id:      "t1",
+		Status:  evergreen.TaskFailed,
+		Version: "v",
+	}
+	assert.NoError(t, t1.Insert())
+	t2 := Task{
+		Id:      "t2",
+		Status:  evergreen.TaskSucceeded,
+		Version: "v",
+	}
+	assert.NoError(t, t2.Insert())
+	dt := Task{
+		Id:                      "dt",
+		DisplayOnly:             true,
+		ExecutionTasks:          []string{"t1", "t2"},
+		Status:                  evergreen.TaskFailed,
+		Version:                 "v",
+		ResetFailedWhenFinished: true,
+	}
+	assert.NoError(t, dt.Insert())
+
+	checkTaskIsArchived := func(t *testing.T, oldTaskID string) {
+		dbTask, err := FindOneOldId(oldTaskID)
+		require.NoError(t, err)
+		require.NotZero(t, dbTask)
+		assert.NotZero(t, dbTask.OldTaskId)
+		assert.NotEqual(t, dbTask.OldTaskId, dbTask.Id)
+		assert.True(t, dbTask.Archived)
+		assert.False(t, dbTask.Aborted)
+		assert.Zero(t, dbTask.AbortInfo)
+	}
+
+	checkTaskIsNotArchived := func(t *testing.T, taskID string, execution int) {
+		task, err := FindOneIdAndExecution(taskID, execution)
+		assert.NoError(t, err)
+		assert.False(t, task.Archived)
+
+		oldT, err := FindOneOldId(MakeOldID(taskID, execution))
+		assert.NoError(t, err)
+		assert.Nil(t, oldT)
+
+		nextExecution, err := FindOneIdAndExecution(taskID, execution+1)
+		assert.NoError(t, err)
+		assert.Nil(t, nextExecution)
+	}
+
+	checkEventLogHostTaskExecutions := func(t *testing.T, hostID, oldTaskID string, execution int) {
+		dbTask, err := FindOneOldId(oldTaskID)
+		require.NoError(t, err)
+		require.NotZero(t, dbTask)
+
+		events, err := event.FindAllByResourceID(hostID)
+		require.NoError(t, err)
+		assert.NotEmpty(t, events)
+
+		for _, e := range events {
+			hostEventData, ok := e.Data.(*event.HostEventData)
+			require.True(t, ok)
+			require.Equal(t, hostEventData.TaskId, dbTask.OldTaskId)
+			require.Equal(t, hostEventData.Execution, strconv.Itoa(dbTask.Execution), len(events))
+		}
+	}
+
+	t.Run("ArchivesOnlyFailedExecutionTasks", func(t *testing.T) {
+		dt.ResetFailedWhenFinished = true
+
+		// Gets the future archived tasks information.
+		t1, err := FindOneIdAndExecution(dt.ExecutionTasks[0], dt.Execution)
+		require.NoError(t, err)
+		t2, err := FindOneIdAndExecution(dt.ExecutionTasks[1], dt.Execution)
+		require.NoError(t, err)
+		archivedT1 := MakeOldID(t1.Id, t1.Execution)
+		archivedExecution := t1.Execution
+
+		hostID := "hostID"
+		event.LogHostRunningTaskSet(hostID, t1.Id, 0)
+		event.LogHostRunningTaskCleared(hostID, t1.Id, 0)
+
+		// Verifies the execution before and after calling Archive
+		archivedDisplayTaskID := MakeOldID(dt.Id, dt.Execution)
+		require.Equal(t, 0, dt.Execution)
+		require.NoError(t, dt.Archive())
+		dtPointer, err := FindOneId(dt.Id)
+		dt = *dtPointer
```

Comment:
There's no reason to do this; execution can be checked directly from what's returned at 3880